benchexec/tablegenerator/react-table/src/components/StatisticsTable.js

```diff
@@ -5,89 +5,132 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
-import React, { useState, useMemo, useEffect } from "react";
-
+import React, { useMemo } from "react";
 import {
-  useTable,
   useFilters,
-  useResizeColumns,
   useFlexLayout,
+  useResizeColumns,
+  useTable,
 } from "react-table";
+import { StandardColumnHeader } from "./TableComponents";
 import { useSticky } from "react-table-sticky";
 import {
-  createRunSetColumns,
-  SelectColumnsButton,
-  StandardColumnHeader,
-} from "./TableComponents";
-import { computeStats, statisticsRows } from "../utils/stats.js";
-import {
-  determineColumnWidth,
-  isNumericColumn,
   isNil,
+  isNumericColumn,
   getHiddenColIds,
+  determineColumnWidth,
 } from "../utils/utils";
 
-const isTestEnv = process.env.NODE_ENV === "test";
-
-const titleColWidth = window.innerWidth * 0.15;
-
 const renderTooltip = (cell) =>
   Object.keys(cell)
     .filter((key) => cell[key] && key !== "sum")
     .map((key) => `${key}: ${cell[key]}`)
     .join(", ") || undefined;
 
-const StatisticsTable = ({
-  selectColumn,
-  tools,
-  switchToQuantile,
-  hiddenCols,
-  tableData,
-  onStatsReady,
-  stats: defaultStats,
-  filtered = false,
-}) => {
-  // We want to skip stat calculation in a test environment if not
-  // specifically wanted (signaled by a passed onStatsReady callback function)
-  const skipStats = isTestEnv && !onStatsReady;
+const StatisticsTable = ({ switchToQuantile, tableData, hiddenCols }) => {
+  const { runSet, runSetStats, runSetIndex } = tableData;
 
-  // When filtered, initialize with empty statistics until computed statistics
-  // are available in order to prevent briefly showing the wrong statistics.
-  const [stats, setStats] = useState(filtered ? [] : defaultStats);
-  const [isTitleColSticky, setTitleColSticky] = useState(true);
+  const columns = useMemo(() => {
+    const createColumnBuilder =
+      ({ switchToQuantile }) =>
+      (column, columnIdx) => ({
+        id: `${runSetIndex}_${column.display_title}_${columnIdx}`,
+        Header: (
+          <StandardColumnHeader
+            column={column}
+            className="header-data clickable"
+            title="Show Quantile Plot of this column"
+            style={{
```

Comment:
Please move CSS to the stylesheets (also below). Not only is it more manageable there, but it also reduces the size of our snapshot tests.